src/org/zaproxy/zap/spider/Spider.java

```diff
@@ -141,11 +145,31 @@
 	 * @param connectionParam the connection param
 	 * @param model the model
 	 * @param scanContext if a scan context is set, only URIs within the context are fetched and processed
+	 * @deprecated (TODO add version) Use {@link #Spider(String, ExtensionSpider, SpiderParam, ConnectionParam, Model, Context)}
+	 *             instead.
 	 */
+	@Deprecated
 	public Spider(ExtensionSpider extension, SpiderParam spiderParam, ConnectionParam connectionParam,
 			Model model, Context scanContext) {
+		this("?", extension, spiderParam, connectionParam, model, scanContext);
+	}
+
+	/**
+	 * Constructs a {@code Spider} with the given data.
+	 * 
+	 * @param id the ID of the spider, usually a unique integer
```

Comment:
As above, does this need to be more clear that it's a string representation not an integer